src/client/datascience/data-viewing/data-wrangler/dataWrangler.ts

```diff
@@ -0,0 +1,821 @@
+                supportsDataExplorer: true,
+                type: 'DataFrame',
+                size: 0,
+                shape: '',
+                count: 0,
+                truncated: true
+            },
+            notebook
+        );
+        const jupyterVariableDataProvider = await this.dataProviderFactory.create(jupyterVariable);
+        // Set dependencies for jupyterVariableDataProvider
+        jupyterVariableDataProvider.setDependencies(jupyterVariable, notebook);
+        // Get variable info
+        this.dataFrameInfoPromise = jupyterVariableDataProvider.getDataFrameInfo();
+        this.dataProvider = jupyterVariableDataProvider;
+        const dataFrameInfo = await this.dataFrameInfoPromise;
+
+        this.postMessage(DataViewerMessages.InitializeData, dataFrameInfo).ignoreErrors();
+    }
+
+    protected async onViewStateChanged(args: WebViewViewChangeEventArgs) {
+        if (args.current.active && args.current.visible && args.previous.active && args.current.visible) {
+            await this.globalMemento.update(PREFERRED_VIEWGROUP, this.webPanel?.viewColumn);
+        }
+        this._onDidChangeDataWranglerViewState.fire();
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    protected onMessage(message: string, payload: any) {
+        let handled = false;
+        switch (message) {
+            case DataWranglerMessages.SubmitCommand:
+                this.handleCommand(payload).ignoreErrors();
+                handled = true;
+                break;
+
+            case DataWranglerMessages.RefreshDataWrangler:
+                this.refreshData().ignoreErrors();
+                handled = true;
+                break;
+
+            case CssMessages.GetMonacoThemeRequest:
+                void this.handleMonacoThemeRequest(payload);
+                handled = true;
+                break;
+
+            default:
+                break;
+        }
+
+        if (!handled) {
+            // Some messages will be handled by DataViewer
+            super.onMessage(message, payload);
+        }
+    }
+
+    private addToHistory(newHistoryItem: IHistoryItem) {
+        this.historyList.push(newHistoryItem);
+        this.postMessage(DataWranglerMessages.UpdateHistoryList, this.historyList).ignoreErrors();
+    }
+
+    private getCode() {
+        return this.historyList.map((item) => item.code).join('\n');
+    }
+
+    private async exportToCsv(_req: undefined, currentVariableName: string) {
+        const notebook = (this.dataProvider as IJupyterVariableDataProvider).notebook;
+        const fileInfo = await this.applicationShell.showSaveDialog({
+            saveLabel: DataScience.dataWranglerSaveCsv(),
+            filters: { CSV: ['csv'] }
+        });
+        if (fileInfo) {
+            const code = `${currentVariableName}.to_csv(path_or_buf=r'${fileInfo.fsPath}', index=False)`;
+            await notebook?.execute(code, '', 0, uuid(), undefined, false);
```

Comment:
I would recommend writing a helper over this function, since you really only care about the `code` part as a parameter